packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/ModelReaderWriterValidation/ModelOperatorsTests.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using UnbrandedTypeSpec.Models;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Tests.ModelReaderWriterValidation
+{
+    public class ModelOperatorsTests
+    {
+        [Test]
+        public async Task TestBinaryContent_Friend()
```

Comment:
lets add a cast round trip strategy and have this validation happen as part of this PR https://github.com/microsoft/typespec/pull/3749

Reply:
That PR was already merged in. I've gone ahead and added the requested tests as part of this PR.